src/libfsm/determinise.c

```diff
@@ -450,9 +460,34 @@ determinise(struct fsm *nfa,
 	 * This is not yet "done"; it starts off the loop below.
 	 */
 	{
+		const struct fsm_state *nfastart;
 		struct fsm_state *dfastart;
+		int includeself = 1;
+
+		nfastart = fsm_getstart(nfa);
+
+		/*
+		 * As a special case for Brzozowski's algorithm, fsm_determinise() is
+		 * expected to produce a minimal DFA for its invocation after the second
+		 * reversal. Since we do not provide multiple start states, fsm_reverse()
+		 * may introduce a new start state which transitions to several states.
+		 * This is the situation we detect here.
+		 *
+		 * This fabricated start state is then excluded from its epsilon closure,
+		 * so that the closures for its destination states are found equivalent,
+		 * because they also do not include the start state.
+		 *
+		 * If you pass an equivalent NFA where this is not the case (for example,
+		 * with the start state containing an epsilon edge to itself), we regard
+		 * this as any other DFA, and minimal output is not guaranteed.
+		 */
```

Comment:
The comment looks good to me.  Possibly worth including a reference to the StackExchange answer https://cs.stackexchange.com/questions/6138/problem-with-implementing-brzozowskis-algorithm?noredirect=1&lq=1

Reply:
I saw that! The answer was too terse for me to follow. So I don't think it'd suit this audience, either.